### ISSUE TITLE
[server crash][infinite loop]Reflection amulet does not play well with Ars Nouveau's Wilden Guardian

### ISSUE THREAD

**Issue by macks2008** [SSKirillSS/relics#17]
Pretty sure I just found a bug between you and Ars Nouveau. Judging by my server's stacktrace, the reflection necklace is getting stuck trying to calculate damage and stalling the whole thread whenever a player wearing one gets attacked by one of AN's "Wilden Guardians"
A sample of the stacktrace (as implied by the matching start & end lines, it repeats these calls endlessly. Stack go brrrrrrrrrrr... `StackOverflowError`!)
```
    at com.hollingsworth.arsnouveau.common.entity.WildenGuardian.func_70665_d(WildenGuardian.java:147)
    at net.minecraft.entity.LivingEntity.func_70097_a(LivingEntity.java:978)
    at net.minecraft.entity.monster.MonsterEntity.func_70097_a(SourceFile:73)
    at it.hurts.sskirillss.relics.items.relics.ReflectionNecklaceItem$ReflectionNecklaceServerEvents.onEntityHurt(ReflectionNecklaceItem.java:134)
    at net.minecraftforge.eventbus.ASMEventHandler_290_ReflectionNecklaceServerEvents_onEntityHurt_LivingHurtEvent.invoke(.dynamic)
    at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:85)
    at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302)
    at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283)
    at net.minecraftforge.common.ForgeHooks.onLivingHurt(ForgeHooks.java:362)
    at net.minecraft.entity.player.PlayerEntity.func_70665_d(PlayerEntity.java:871)
    at net.minecraft.entity.LivingEntity.func_70097_a(LivingEntity.java:978)
    at net.minecraft.entity.player.PlayerEntity.func_70097_a(PlayerEntity.java:814)
    at net.minecraft.entity.player.ServerPlayerEntity.func_70097_a(ServerPlayerEntity.java:597)
    at com.hollingsworth.arsnouveau.common.entity.WildenGuardian.func_70665_d(WildenGuardian.java:147)
```
By the way, nice top level package?

**Comment by bigenergy** [comment]
same problem

**Comment by SSKirillSS** [comment]
Hi. This issue should be fixed in https://github.com/SSKirillSS/relics/commit/21f4b2b6bb23ea4a39dd3578cd1dbac4d5d15a83.

Btw you still should wait for the next update (or you can just implement this patch to your own version if you know how to do it).

**Comment by macks2008** [comment]
yup, I can confirm that it seems to be fixed in the latest build.
I would just use the version you gave me, except I won't be able to share it with my friends through my curseforge pack until you push the update because of how strict curseforge is about pack overrides (basically, any file included in the packet has to either be on the cursed CDN, or the mod has to give explicit permission for overrides... or something like that) :-(
let me know, preferably with a ping either here or better yet on your discord, when you push that version. In the meantime I've warned my friends not to wear a reflection amulet in the presence of a guardian... none of them are cruel enough to me to crash the server deliberately, thankfully.

**Comment by SSKirillSS** [comment]
Should be fixed in the 0.2.0 update.
